src/test/java/io/vantiq/client/intg/VantiqIntegrationTest.java

```diff
@@ -630,7 +719,19 @@ public void testAcknowledgeReliableTopic() throws Exception {
 
         // Subscribe to topic
         vantiq.subscribe(Vantiq.SystemResources.TOPICS.value(), "/test/topic", null, callback, params);
-        callback.waitForCompletion();
+        // These types of subscriptions may be a multi-step process where by > 1 message is required to complete the
+        // connection.  callback.waitForCompletion() just waits until some message arrives, not necessarily the
+        // connection message.  callback.waitForConnection() will loop (up to the specified (or default) timeout
+        // period until it sees that the connection status is satisfied.  If not (that is, timeout occurs), then it
```

Comment:
period)

Reply:
I'm having flashbacks to working on the university postgres system which was ported from Allegro Common Lisp to C. Picture a query plan constructed as a lisp list. car, cdr, cadr, cadddddddr C macros were everywhere--it was awesome.